### PR TITLE
chore: Add test for create2 

### LINKED ISSUE #7558: Add test for create2 deploy merging into hollow account and self destructing
Add an EET for the following scenario:
1. create a hollow account at address create2address
2. subsequent transfers to create2address result in account balance updates as expected
3. deploy a contract with CREATE2 at the create2address and make that contract self-destruct in its constructor
4. if we transfer to address create2address a new hollow account is created with the create2address alias
5. if we redeploy another instance of the contract at the same create2address, a new contract is deployed with another entity id

### PR DESCRIPTION
**Pull request by stoqnkpL**
**Description**:

Adds an EET for the following scenario:

1. create a hollow account at address create2address
2. subsequent transfers to create2address result in account balance updates as expected
3. deploy a contract with CREATE2 at the create2address and make that contract self-destruct in its constructor
4. if we transfer to address create2address a new hollow account is created with the create2address alias
5. if we redeploy another instance of the contract at the same create2address, a new contract is deployed with another entity id

Adds an EET for the following scenario:

1. create aliased account by transferring hbar to an EVM address alias
2. send an EthereumTransaction from the account, increasing the account nonce to 1
3. delete the account
4. attempt transferring hbar to the EVM address alias; expect response code `ACCOUNT_DELETED`

**Related issue(s)**:

Fixes #7558 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Comment by Nana-EC** [comment]
@stoqnkpL should we rebase this and get it in?

**Review by Nana-EC — DISMISSED** [review]

File: hedera-node/test-clients/src/main/resource/contract/contracts/Create2FactoryWithSelfDestructingContract/Create2FactoryWithSelfDestructingContract.sol [hunk 1]
```diff
@@ -0,0 +1,70 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.9;
+
+contract Create2FactoryWithSelfDestructingContract {
+    event Deployed(address addr, uint salt);
+
+    // 1. Get bytecode of contract to be deployed
+    // NOTE: _owner and _foo are arguments of the SelfDestructingTestContract's constructor
+    function getBytecode(address _owner, uint _foo) public pure returns (bytes memory) {
```
Comment (lukelee-sl) [inline hedera-node/test-clients/src/main/resource/contract/contracts/Create2FactoryWithSelfDestructingContract/Create2FactoryWithSelfDestructingContract.sol:9]: Nit:  I understand _foo is just a random uint but maybe a better name would be good?

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyContractTestsSuite.java [hunk 2]
```diff
@@ -295,7 +310,105 @@ public List<HapiSpec> getSpecsInSuite() {
...
+                                .payingWith(GENESIS)
+                                .gas(4_000_000L)
+                                .sending(tcValue)
+                                .via(CREATE_2_TXN_2)),
+                        captureOneChildCreate2MetaFor(
+                                "Merged deployed contract with hollow account and self-destructed the contract",
+                                CREATE_2_TXN_2,
+                                mergedMirrorAddr2,
+                                mergedAliasAddr2),
+                        sourcing(() -> getContractInfo(mergedMirrorAddr2.get())
```
Comment (lukelee-sl) [inline hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyContractTestsSuite.java:402]: Not super important but is there a way to assert that the hbar used to create the hollow account went back to the contract that created the contract which was self-destructed?

**Review by lukelee-sl — DISMISSED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyContractTestsSuite.java (hunk 2 above)
Comment (stoqnkpL) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/leaky/LeakyContractTestsSuite.java:402]: That check is done on the previous self destruct:
First on line 375 we assert that `contract` has 0 balance:
```
getContractInfo(contract)
                                .has(ContractInfoAsserts.contractWith().balance(0L)),
```
And then, after the inner contract deploy and self-destruct, on line 389 we assert that `contract` now has a positive balance equal to the sum of the initial HBAR transfer that created the account and the value that was transferred to the contract as part of the function call:
```
getContractInfo(contract)
                            .has(ContractInfoAsserts.contractWith().balance(ONE_HBAR + tcValue)),
```

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Review by Nana-EC — DISMISSED** [review]

**Comment by Nana-EC** [comment]
@stoqnkpL moved from milestone 0.41 to 0.46.
Please modify if this isn't correct

**Review by tinker-michaelj — APPROVED** [review]
LGTM, tyvm @stoqnkpL !

**Pull request merged**